apps/google-analytics-4/frontend/src/hooks/useKeyService.tsx

```diff
@@ -37,60 +39,71 @@ export default function useKeyService(props: Props): KeyServiceInfoType {
   const [serviceAccountKeyFileIsRequired, setServiceAccountKeyFileIsRequired] =
     useState<boolean>(false);
 
-  const sdk = useSDK<AppExtensionSDK>();
-
-  const onConfigure = useCallback(async () => {
-    const currentState = await sdk.app.getCurrentState();
+  const sdk = useSDK<any>();
 
-    if (!serviceAccountKeyFileIsValid) {
-      sdk.notifier.error('Invalid service account key file. See field error for details');
-      return false;
-    }
+  init((sdkLocation) => {
+    if (sdkLocation.location.is(locations.LOCATION_APP_CONFIG)) setIsConfigureScreen(true);
+    else setIsConfigureScreen(false);
+  });
```

Comment:
All of these changes in this file will likely be adjusted with a refactor of this service. Just needed to add some conditional statements to assure this works in the config screen AND the side bar.

Reply:
Yes, this will definitely need to be refactored but looks good for now!